python/plugins/processing/algs/qgis/Rasterize.py

```diff
@@ -0,0 +1,282 @@
+        self.settings.setOutputDpi(self.image.logicalDpiX())
+        self.settings.setOutputImageFormat(QImage.Format_RGB32)
+        self.settings.setDestinationCrs(crs)
+        self.settings.setOutputSize(self.image.size())
+        self.settings.setFlag(QgsMapSettings.Antialiasing, True)
+        self.settings.setFlag(QgsMapSettings.RenderMapTile, True)
+
+        if QgsProject.instance().mapThemeCollection().hasMapTheme(map_theme):
+            self.settings.setLayers(
+                QgsProject.instance().mapThemeCollection(
+
+                ).mapThemeVisibleLayers(
+                    map_theme))
+            self.settings.setLayerStyleOverrides(
+                QgsProject.instance().mapThemeCollection(
+
+                ).mapThemeStyleOverrides(
+                    map_theme))
+        elif layer:
+            self.settings.setLayers([layer])
+        else:
+            self.settings.setLayers(map_settings.layers())
+
+    def render(self):
+        for x in range(self.x_tile_count):
+            for y in range(self.y_tile_count):
+                cur_tile = x * self.y_tile_count + y
+                num_tiles = self.x_tile_count * self.y_tile_count
+                self.renderTile(x, y)
+
+    def renderTile(self, x, y):
+        """
+        Render one tile
+
+        :param x: The x index of the current tile
+        :param y: The y index of the current tile
+        """
+        painter = QPainter(self.image)
+
+        self.settings.setExtent(QgsRectangle(
+            self.extent.xMinimum() + x * self.mupp * self.tile_size,
+            self.extent.yMaximum() - (y + 1) * self.mupp * self.tile_size,
+            self.extent.xMinimum() + (x + 1) * self.mupp * self.tile_size,
+            self.extent.yMaximum() - y * self.mupp * self.tile_size))
+
+        job = QgsMapRendererCustomPainterJob(self.settings, painter)
+        job.renderSynchronously()
+        painter.end()
+
+        # Needs not to be deleted or Windows will kill it too early...
+        tmpfile = tempfile.NamedTemporaryFile(suffix='.png', delete=False)
+        try:
+            self.image.save(tmpfile.name)
+
+            src_ds = osgeo.gdal.Open(tmpfile.name)
+
+            self.dataset.WriteRaster(x * self.tile_size, y * self.tile_size,
+                                     self.tile_size, self.tile_size,
+                                     src_ds.ReadRaster(0, 0, self.tile_size,
+                                                       self.tile_size))
+        finally:
+            del src_ds
+            tmpfile.close()
+            os.unlink(tmpfile.name)
+
+    def getDriverForFile(self, filename):
+        """
+        Get the GDAL driver for a filename, based on its extension. (.gpkg, .mbtiles...)
+        """
+        _, extension = os.path.splitext(filename)
+
+        # If no extension is set, use .tif as default
+        if extension == '':
+            extension = '.tif'
+
+        for i in range(osgeo.gdal.GetDriverCount()):
```

Comment:
@alexbruy is this the correct way to get the driver or is there a standard processing method?